pow/src/main/java/tech/pegasys/teku/pow/contract/DepositContract.java

```diff
@@ -138,11 +138,19 @@ protected DepositContract(
             .ethGetLogs(filter)
             .sendAsync()
             .thenApply(
-                logs ->
-                    logs.getLogs().stream()
-                        .map(log -> (Log) log.get())
-                        .map(this::convertLogToDepositEventEventResponse)
-                        .collect(Collectors.toList())));
+                logs -> {
+                  if (logs.getLogs() == null) {
+                    // We got a response from the node but it didn't include even an empty list
+                    // of logs.  This happens with Infura when more than 10,000 log entries match
+                    // so treat as an explicit rejection of the request to allow the requested block
+                    // range to be reduced.
+                    throw new RejectedRequestException("No logs returned by ETH1 node");
```

Comment:
'FROM' eth1 node?

Reply:
Grammatically I'd have said `by` - ie it was the ETH1 node that produced the empty response.  It really doesn't matter though because we catch the exception and handle it anyway.